### PR TITLE
Implement support for structured stack traces

### PR DESCRIPTION
**Pull request by ofrobots**
PTAL @matthewloring

### CODE REVIEW

File: lib/span-data.js [hunk 1]
```diff
@@ -82,6 +92,23 @@ SpanData.prototype.close = function() {
...
 /**
+ * Trace API expects stack frames to be a JSON string with the following
+ * structure:
+ * STACK_TRACE := { "stack_frame" : [ FRAMES ] }
+ * FRAMES := { "class_name" : CLASS_NAME, "file_name" : FILE_NAME,
+ *             "line_number" : LINE_NUMBER, "method_name" : METHOD_NAME }*
+ * @constructor @private
+ */
+function StackFrame(className, methodName, fileName, lineNumber, columnNumber) {
+  if (className)   { this.class_name = className;  }
```
Comment (matthewloring) [inline lib/span-data.js]: Why are these if's necessary? Does the visible behavior change if they are not included?

**Comment by matthewloring** [comment]
Fixes #19

File: lib/span-data.js (hunk 1 above)
Comment (ofrobots) [inline lib/span-data.js]: My thinking was that I want to leave the property undefined, as opposed to being set to null, if null is passed in; this reduces the data sent on the network. For example, className will always be null in our case. Upon reflection, this does the wrong thing for lineNumber or columnNumber when they are 0. I'll try to fix this.

**Comment by ofrobots** [comment]
@matthewloring ready for another look.

File: lib/span-data.js [hunk 2]
```diff
@@ -82,6 +92,42 @@ SpanData.prototype.close = function() {
 };
 
 /**
+ * Trace API expects stack frames to be a JSON string with the following
+ * structure:
+ * STACK_TRACE := { "stack_frame" : [ FRAMES ] }
+ * FRAMES := { "class_name" : CLASS_NAME, "file_name" : FILE_NAME,
+ *             "line_number" : LINE_NUMBER, "method_name" : METHOD_NAME }*
```
Comment (matthewloring) [inline lib/span-data.js]: Nit: could we include columnNumber in this graphic as it is a member of the class.

File: lib/span-data.js [hunk 3]
```diff
@@ -82,6 +92,42 @@ SpanData.prototype.close = function() {
...
+ * @param {string|undefined} className
+ * @param {string|undefined} methodName
+ * @param {string|undefined} fileName
+ * @param {number|undefined} lineNumber
+ * @param {number|undefined} columnNumber
+ * @constructor @private
+ */
+function StackFrame(className, methodName, fileName, lineNumber, columnNumber) {
+  if (className) {
+    this.class_name = className;
```
Comment (matthewloring) [inline lib/span-data.js]: Note to future me: This path will be dead code as we currently do not provide class names. It is included to be faithful to the structure expected by the server.

**Comment by matthewloring** [comment]
LGTM.

**Pull request merged**